trpcultivate_phenotypes/src/PhenodataBackupListBuilder.php

```diff
@@ -0,0 +1,339 @@
+   */
+  protected EntityTypeManagerInterface $service_EntityTypeManager;
+
+  /**
+   * Users.
+   *
+   * @var \Drupal\Core\Session\AccountInterface
+   */
+  protected AccountInterface $user;
+
+  /**
+   * Configuration entity fields per permission.
+   *
+   * @var array
+   */
+  private $entity_field_header = [];
+
+  /**
+   * Constructor.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeInterface $entity_type
+   *   The entity type definition.
+   * @param \Drupal\Core\Entity\EntityStorageInterface $storage
+   *   The entity storage class.
+   * @param \Drupal\Core\From\FormBuilderInterface $form_builder
+   *   The form builder interface.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   * @param \Drupal\Core\Session\AccountInterface $user
+   *   Drupal users.
+   */
+  public function __construct(
+    EntityTypeInterface $entity_type,
+    EntityStorageInterface $storage,
+    FormBuilderInterface $form_builder,
+    EntityTypeManagerInterface $entity_type_manager,
+    AccountInterface $user,
+  ) {
+    parent::__construct($entity_type, $storage);
+
+    $this->service_EntityTypeManager = $entity_type_manager;
+    $this->formBuilder = $form_builder;
+    $this->user = $user;
+
+    $headers = [
+      'project_id' => $this->t('Project/Experiment'),
+      'comments' => $this->t('Notes/Comments'),
+      'backup_date' => $this->t('Date Created'),
+      'file_id' => $this->t('Data File'),
+      'user_id' => $this->t('Created By'),
+    ];
+
+    $has_view_all = $this->user->hasPermission('view_all phenodata_backup');
+    $has_view_own = $this->user->hasPermission('view_own phenodata_backup');
+
+    if ($has_view_all || $this->user->hasPermission('administer phenodata_backup')) {
+      $this->entity_field_header = $headers;
+    }
+    elseif ($has_view_own) {
+      unset($headers['user_id']);
+      $this->entity_field_header = $headers;
+    }
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public static function createInstance(ContainerInterface $container, EntityTypeInterface $entity_type) {
+
+    return new static(
+      $entity_type,
+      $container->get('entity_type.manager')->getStorage($entity_type->id()),
+      $container->get('form_builder'),
+      $container->get('entity_type.manager'),
+      $container->get('current_user'),
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildHeader(): array {
+
+    return $this->entity_field_header + parent::buildHeader();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildRow(EntityInterface $entity): array {
+
+    $values = [];
+
+    $key = 'project_id';
+    $project_id = (int) $entity->get($key);
+    $project_name = ChadoProjectAutocompleteController::getProjectName($project_id);
+    $values[$key]['data'] = [
+      '#markup' => '<a target="_blank" href="/experiment/' . str_replace(' ', '-', $project_name) . '">' . $project_name . '</a>',
+    ];
```

Comment:
This makes an assumption on the url for the experiment which may not always hold true :grimaces. Instead we actually need to get the entity id for the project and check that the user has permission to it before showing it as a link. Additionally, you should use a link render array here rather then `#markup`.

Reply:
This has been moved to #154. Instead for this PR lets just use the project name and not make it a link. :-)